bigchaindb/common/election.py

```diff
@@ -0,0 +1,248 @@
+# Copyright BigchainDB GmbH and BigchainDB contributors
+# SPDX-License-Identifier: (Apache-2.0 AND CC-BY-4.0)
+# Code is Apache-2.0 and docs are CC-BY-4.0
+
+import base58
+
+from bigchaindb import backend
+from bigchaindb.common.exceptions import (InvalidSignature,
+                                          MultipleInputsError,
+                                          InvalidProposer,
+                                          UnequalValidatorSet,
+                                          DuplicateTransaction)
+from bigchaindb.tendermint_utils import key_from_base64
+from bigchaindb.common.crypto import (public_key_from_ed25519_key)
+from bigchaindb.common.transaction import Transaction
+from bigchaindb.common.schema import (_validate_schema,
+                                      TX_SCHEMA_COMMON,
+                                      TX_SCHEMA_CREATE)
+
+
+class Election(Transaction):
+
+    # NOTE: this transaction class extends create so the operation inheritance is achieved
+    # by setting an ELECTION_TYPE and renaming CREATE = ELECTION_TYPE and ALLOWED_OPERATIONS = (ELECTION_TYPE,)
+    ELECTION_TYPE = None
```

Comment:
In an attempt to make the code more uniform it would good to rename `ELECTION_TYPE` to `OPERATION` so that further in future when `CREATE` and `TRANSFER` both become an independent class they would also have `OPERATION` field and the registeration would look as follows,

```
Transaction.register_type(Create.OPERATION, models.Transaction)
Transaction.register_type(Transfer.OPEATION, models.Transaction)
Transaction.register_type(ValidatorElection.OPERATION, ValidatorElection)
Transaction.register_type(ValidatorElectionVote.OPERATION, ValidatorElectionVote)
```

Reply:
Done